lua/gitlab/reviewer/diffview.lua

```diff
@@ -16,7 +16,15 @@ M.open = function()
     return
   end
 
-  vim.api.nvim_command(string.format("DiffviewOpen %s..%s", diff_refs.base_sha, diff_refs.head_sha))
```

Comment:
This was not really about the notification itself but rather about how we open diffview nvim. What I mean is that currently ( master branch ) if I open reviewer I see these buffers for diffview 
```
 11u#a-  "diffview://null"              line 1                                                                                                                                                                                              
 12u%a-  "diffview:///panels/0/DiffviewFilePanel" line 5                                                                                                                                                                                    
 13u a-  "diffview:///home/git/dummy-test-repo/.git/71f706c3cf0/test.py" line 0 
``` 
and LSP is not available during review in review panel -> you cannot check definition or jump to definition.

When I check out this branch (reviewer_improvement) then I see these buffers for diffview
```
 11u#a-  "diffview://null"              line 1                                                                                                                                                                                              
 12u%a-  "diffview:///panels/0/DiffviewFilePanel" line 5                                                                                                                                                                                    
 13  a   "~/dummy-test-repo/test.py" line 0
``` 
as you can see the buffer 13 is pointing to real file ( test.py is new file that is why buffer 11 is `diffview://null` ) and all LSP commands are available and language server runs for that file so while reviewing new code I can actually see definitions, diagnostics etc. We already require user to be in correct branch so I thought that it would be enough to have warning ( or even don't open reviewer as you wrote ) to notify user about commit missmatch and still open against current files with `DiffviewOpen %s`. If you don't like the idea can I make it at least configurable ? I consider having LSP during review a huge benefit of this plugin.

Reply:
@harrisoncramer any thoughts on this ?

Reply:
This doesn't work if either commit lines up, for instance see this issue:

https://github.com/harrisoncramer/gitlab.nvim/issues/63

When reviewing against the local filesystem you do not actually have any guarantees that you are actually reviewing the changes in the MR itself. You would need to check that _both_ the feature branch commit SHA _and_ the target branch commit SHA match those from Gitlab. I don't think this should be the default behavior since it's going to be confusing for the vast majority of users, who do not want to adjust their local filesystem to review someone else's MR. 

If you wanted to make it an option, you'd have to check both commits. If they don't match we should flash an error and quit, and force the user to adjust their filesystem to match. The error message should also suggest switching to the default behavior for folks who add this option by accident, something like:

"Your commit SHAs on %target branch% and %feature branch% do not match, they should be %X% and %Y*%. Please adjust them or remove the `use_local_filesystem` option from your setup call"

Feel pretty strongly though that this approach should be discouraged for most users though.

Reply:
It should be enough to check only if `diff_refs.head_sha` is matched with current `HEAD` if we open reviewer against specific commit like this `vim.api.nvim_command(string.format("DiffviewOpen %s", diff_refs.base_sha))` - this should always result in correct base commit ( if commit is available on local filesystem).

Sure I agree that this don't have to be default and that the documentation should state downsides.

I will try to implement the change with optional setting, documentation and will try to add check that current commit is same as `diff_refs.head_sha` and also that there are no additional changes against `diff_refs.head_sha`.

Then you can decide if you want to merge it or not :+1: ( but I would love to have that option )

Reply:
Sounds good.

Reply:
Hi,
I added new option `open_reviewer_against` to  settings where user  can choose either `commit` ( default) or `filesystem` and also added documentation to this. Let me know what do you think about it.